### ISSUE TITLE
Getting a tensorflow error in the keras build

### ISSUE THREAD

**Issue by ericdill** [conda-forge/tensorflow-feedstock#19]
I'm getting the following error in the test phase of the keras build:

```
===== testing package: keras-1.1.0-py27_0 =====
import: 'keras'
Using TensorFlow backend.
Traceback (most recent call last):
  File "/feedstock_root/build_artefacts/test-tmp_dir/run_test.py", line 26, in <module>
    import keras
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/keras/__init__.py", line 2, in <module>
    from . import backend
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/keras/backend/__init__.py", line 64, in <module>
    from .tensorflow_backend import *
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/keras/backend/tensorflow_backend.py", line 1, in <module>
    import tensorflow as tf
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/tensorflow/__init__.py", line 23, in <module>
    from tensorflow.python import *
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/tensorflow/python/__init__.py", line 48, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 28, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "/opt/conda/envs/_test/lib/python2.7/site-packages/tensorflow/python/pywrap_tensorflow.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)
ImportError: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /opt/conda/envs/_test/lib/python2.7/site-packages/tensorflow/python/_pywrap_tensorflow.so)
TESTS FAILED: keras-1.1.0-py27_0
```

Is there a known fix for this?

circleci build: https://circleci.com/gh/conda-forge/keras-feedstock/5

**Comment by jjhelmus** [comment]
The current `conda-forge` tensorflow package is a repacking of the whl files provided by Google.  These seem to have been build on a Ubuntu 14.04 (?) system which uses a newer GLIBC that the one on CentOS 6 which is the default `conda-forge` target.  This higher GLIBC version is likely causing the issue you are seeing.    See issue #11 for a longer discussion.

The fix for this is to build tensorflow from source on a CentOS 6 platform.  I started this in PR #12 but tensorflow uses a non-standard build process (at least for non-Googlers).  Additionally, the processes has changed for the 0.11.0rc0 release, #17, which needs protobuf 3.0.0, see conda-forge/protobuf-feedstock#13 which is also proving difficult to build.  Any help on these would be much appreciated.

As a workaround for the time being you could try using the `tensorflow` package from my channel, https://anaconda.org/jjhelmus/tensorflow, which was build from source in a CentOS 6 docker container using the [instructions and recipe in my `wip_conda_recipes` repo](https://github.com/jjhelmus/wip_conda_recipes/tree/master/tensorflow).

**Comment by ericdill** [comment]
Thanks for the explanation @jjhelmus. Seems like a nasty build issue.  I'll try and help out if time permits

**Comment by ghego** [comment]
This is still not fixed. Anyone working on it?

**Comment by jjhelmus** [comment]
> This is still not fixed. Anyone working on it?

I have not had the time to work on this and do not expect to in the near future.

**Comment by xhochy** [comment]
This should be fixed with #110
